lib/wallet/src/Cardano/Wallet/DB/Store/Transactions/Model.hs

```diff
@@ -105,40 +105,40 @@ data TxRelation =
     }
     deriving ( Generic, Eq, Show )
 
--- | Transactions history is 'TxRelation's indexed by 'TxId'
-newtype TxHistory =
-    TxHistory { relations :: Map TxId TxRelation }
+-- | A 'TxPile' is a map of 'TxRelation's indexed by their 'TxId'.
+newtype TxPile =
+    TxPile { relations :: Map TxId TxRelation }
     deriving ( Generic, Eq, Show )
 
-instance Monoid TxHistory where
-    mempty = TxHistory mempty
+instance Monoid TxPile where
+    mempty = TxPile mempty
 
-instance Semigroup TxHistory where
-    TxHistory h1 <> TxHistory h2 =
-        TxHistory $ h1 <> h2
+instance Semigroup TxPile where
+    TxPile h1 <> TxPile h2 =
+        TxPile $ h1 <> h2
```

Comment:
Suggestion: derive these instances instead of defining them by hand:
```patch
  newtype TxPile =
      TxPile { relations :: Map TxId TxRelation }
      deriving (Generic, Eq, Show)
+     deriving newtype (Semigroup, Monoid)
```

Reply:
will tackle this in another PR